SpatialGDK/Source/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -150,19 +161,31 @@ void USpatialWorkerConnection::ConnectToLocator()
 		ConnectionParams.network.use_external_ip = SpatialConnection->LocatorConfig.UseExternalIp;
 		// end TODO
 
+		bool bFoundRequestedDeployment = true;
 		int DeploymentIndex = 0;
 		if (!SpatialConnection->LocatorConfig.DeploymentName.IsEmpty())
 		{
+			bFoundRequestedDeployment = false;
 			for (uint32_t i = 0; i < DeploymentList->deployment_count; ++i)
 			{
 				if (SpatialConnection->LocatorConfig.DeploymentName.Equals(UTF8_TO_TCHAR(DeploymentList->deployments[i].deployment_name)))
 				{
 					DeploymentIndex = i;
+					bFoundRequestedDeployment = true;
 					break;
 				}
 			}
 		}
 
+		if (!bFoundRequestedDeployment)
```

Comment:
This if statement can only be called if the DeploymentName is set in the LocatorConfig. Why not move this into the if section for when a deployment is specified and the loop finished without finding the deployment in the deployment list. that way you can get rid of `bFoundRequestedDeployment`